addons/hr_expense/models/hr_expense_sheet.py

```diff
@@ -657,6 +657,8 @@ def _do_reset_approval(self):
         self.activity_update()
 
     def _do_refuse(self, reason):
+        if any(self.mapped('account_move_ids')):  # Todo: in 17.3+, edit it to allow draft entries"
+            raise UserError(_("You cannot cancel an expense sheet linked to a journal entry"))
         self.write({'state': 'cancel'})
```

Comment:
I think it should be `self.write({'state': 'cancel'})` unlike yours sir

Reply:
calling .write on a single field is a bad practice that hinders readability
We could update update both fields at the same time, but I think it's better to use the compute in case the behaviour changes in the future

Reply:
okay then